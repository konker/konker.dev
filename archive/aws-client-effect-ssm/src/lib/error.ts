export const TAG = 'SsmError';

export type SsmError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSsmError =
  <I>(params: I) =>
  (x: unknown): SsmError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
