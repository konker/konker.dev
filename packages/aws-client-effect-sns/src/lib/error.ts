export const TAG = 'SnsError';

export type SnsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSnsError =
  <I>(params: I) =>
  (x: unknown): SnsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
