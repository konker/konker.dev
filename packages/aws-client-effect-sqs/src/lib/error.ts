export const TAG = 'SqsError';

export type SqsError = {
  readonly _tag: typeof TAG;
  readonly _Params: unknown;
  readonly message: string;
  readonly cause: unknown;
};

export const toSqsError =
  <I>(params: I) =>
  (x: unknown): SqsError => {
    return {
      _tag: TAG,
      _Params: params,
      message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
      cause: x,
    };
  };
