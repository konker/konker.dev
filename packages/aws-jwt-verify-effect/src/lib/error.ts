export const TAG = 'JwtVerifyError';

export type JwtVerifyError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export const toJwtVerifyError = (x: unknown): JwtVerifyError => {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
};
