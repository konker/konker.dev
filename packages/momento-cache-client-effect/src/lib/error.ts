import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'MomentoClientError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type MomentoClientError = TinyError<ERROR_TAG>;
export const toMomentoClientError = toTinyError<ERROR_TAG>(ERROR_TAG);
