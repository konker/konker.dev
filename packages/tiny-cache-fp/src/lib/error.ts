import type { TinyError } from '@konker.dev/tiny-error-fp';
import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'CacheError' as const;
export type ERROR_TAG = typeof ERROR_TAG;

export type CacheError = TinyError<ERROR_TAG>;
export const toCacheError = toTinyError<ERROR_TAG>(ERROR_TAG);
