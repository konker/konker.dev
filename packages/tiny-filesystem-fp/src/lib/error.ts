export const TAG = 'TinyFileSystemError';

export type TinyFileSystemError = {
  readonly _tag: typeof TAG;
  readonly message: string;
  readonly cause: unknown;
};

export function toTinyFileSystemError(x: unknown): TinyFileSystemError {
  return {
    _tag: TAG,
    message: typeof x === 'object' && x && 'message' in x ? (x as any).message : String(x),
    cause: x,
  };
}
