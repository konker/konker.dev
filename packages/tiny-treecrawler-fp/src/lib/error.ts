import { toTinyError } from '@konker.dev/tiny-error-fp';

export const ERROR_TAG = 'TinyTreeCrawlerError';
export type ERROR_TAG = typeof ERROR_TAG;

export const toTinyTreeCrawlerError = toTinyError<ERROR_TAG>(ERROR_TAG);
export type TinyTreeCrawlerError = ReturnType<typeof toTinyTreeCrawlerError>;
