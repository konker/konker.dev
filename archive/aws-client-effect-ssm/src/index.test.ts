import type {
  DeleteParameterCommandInput,
  DeleteParametersCommandInput,
  GetParameterCommandInput,
  GetParametersByPathCommandInput,
  GetParametersCommandInput,
  PutParameterCommandInput,
} from '@aws-sdk/client-ssm';
import * as ssmClient from '@aws-sdk/client-ssm';
import { SSMClient } from '@aws-sdk/client-ssm';
import { mockClient } from 'aws-sdk-client-mock';
import { pipe } from 'effect';
import * as Effect from 'effect/Effect';
import { beforeAll, beforeEach, describe, expect, it } from 'vitest';

import * as unit from './index.js';
import { SSMClientDeps } from './index.js';
import { TAG } from './lib/error.js';

const ssmMock = mockClient(ssmClient.SSMClient);

describe('aws-client-effect-ssm', () => {
  let deps: unit.SSMClientDeps;

  beforeAll(() => {
    deps = unit.SSMClientDeps.of({
      ssmClient: new ssmClient.SSMClient({}),
    });
  });

  describe('Error tag', () => {
    it('should export as expected', () => {
      expect(unit.SSM_ERROR_TAG).toEqual('SsmError');
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSSMClientFactory', () => {
    it('should work as expected', async () => {
      expect(unit.defaultSSMClientFactory({})).toBeInstanceOf(SSMClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSSMClientFactoryDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = pipe(
        unit.SSMClientFactoryDeps,
        Effect.map((deps) => deps.ssmClientFactory),
        unit.defaultSSMClientFactoryDeps
      );
      const actual = Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(Function);
    });
  });

  // ------------------------------------------------------------------------
  describe('defaultSSMClientDeps', () => {
    it('should work as expected', async () => {
      const actualEffect = pipe(
        unit.SSMClientDeps,
        Effect.map((deps) => deps.ssmClient),
        unit.defaultSSMClientDeps({})
      );
      const actual = Effect.runSync(actualEffect);
      expect(actual).toBeInstanceOf(SSMClient);
    });
  });

  // ------------------------------------------------------------------------
  describe('DeleteParameterCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.DeleteParameterCommand).resolves({ $metadata: { httpStatusCode: 200 } });

      const params: DeleteParameterCommandInput = {
        Name: 'test-param-name',
      };
      const expected = { $metadata: { httpStatusCode: 200 }, _Params: params };
      const command = pipe(unit.DeleteParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.DeleteParameterCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: DeleteParameterCommandInput = {
        Name: 'test-param-name',
      };
      const command = pipe(unit.DeleteParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('DeleteParametersCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.DeleteParametersCommand).resolves({ $metadata: { httpStatusCode: 200 } });

      const params: DeleteParametersCommandInput = {
        Names: ['test-param-name-1', 'test-param-name-2'],
      };
      const expected = { $metadata: { httpStatusCode: 200 }, _Params: params };
      const command = pipe(unit.DeleteParametersCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.DeleteParametersCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: DeleteParametersCommandInput = {
        Names: ['test-param-name-1', 'test-param-name-2'],
      };
      const command = pipe(unit.DeleteParametersCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('GetParameterCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.GetParameterCommand).resolves({
        Parameter: { Name: 'test-param-name-1', Type: 'String', Value: 'test-value-1' },
        $metadata: { httpStatusCode: 200 },
      });

      const params: GetParameterCommandInput = {
        Name: 'test-param-name',
      };
      const expected = {
        Parameter: { Name: 'test-param-name-1', Type: 'String', Value: 'test-value-1' },
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = pipe(unit.GetParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.GetParameterCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: GetParameterCommandInput = {
        Name: 'test-param-name',
      };
      const command = pipe(unit.GetParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('GetParametersByPathCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.GetParametersByPathCommand).resolves({
        Parameters: [
          { Name: '/foo/bar/test-param-name-1', Type: 'String', Value: 'test-value-1' },
          { Name: 'test-param-name-2', Type: 'String', Value: 'test-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
      });

      const params: GetParametersByPathCommandInput = { Path: '/foo/bar' };
      const expected = {
        Parameters: [
          { Name: '/foo/bar/test-param-name-1', Type: 'String', Value: 'test-value-1' },
          { Name: 'test-param-name-2', Type: 'String', Value: 'test-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = pipe(unit.GetParametersByPathCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.GetParametersByPathCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: GetParametersByPathCommandInput = { Path: '/foo/bar' };
      const command = pipe(unit.GetParametersByPathCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('GetParametersCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.GetParametersCommand).resolves({
        Parameters: [
          { Name: 'test-param-name-1', Type: 'String', Value: 'test-value-1' },
          { Name: 'test-param-name-2', Type: 'String', Value: 'test-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
      });

      const params: GetParametersCommandInput = {
        Names: ['test-param-name-1', 'test-param-name-2'],
      };
      const expected = {
        Parameters: [
          { Name: 'test-param-name-1', Type: 'String', Value: 'test-value-1' },
          { Name: 'test-param-name-2', Type: 'String', Value: 'test-value-2' },
        ],
        $metadata: { httpStatusCode: 200 },
        _Params: params,
      };
      const command = pipe(unit.GetParametersCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.GetParametersCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: GetParametersCommandInput = { Names: ['test-param-name-1', 'test-param-name-2'] };
      const command = pipe(unit.GetParametersCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });

  // ------------------------------------------------------------------------
  describe('PutParameterCommand', () => {
    beforeEach(() => {
      ssmMock.reset();
    });

    it('should work as expected', async () => {
      ssmMock.on(ssmClient.PutParameterCommand).resolves({
        $metadata: { httpStatusCode: 201 },
      });

      const params: PutParameterCommandInput = {
        Name: 'test-param-name-2',
        Type: 'String',
        Value: 'test-value-2',
      };
      const expected = { $metadata: { httpStatusCode: 201 }, _Params: params };
      const command = pipe(unit.PutParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).resolves.toStrictEqual(expected);
      expect(ssmMock.calls().length).toEqual(1);
    });

    it('should fail as expected', async () => {
      ssmMock.on(ssmClient.PutParameterCommand).rejects({ $metadata: { httpStatusCode: 404 } });

      const params: PutParameterCommandInput = {
        Name: 'test-param-name-2',
        Type: 'String',
        Value: 'test-value-2',
      };
      const command = pipe(unit.PutParameterCommandEffect(params), Effect.provideService(SSMClientDeps, deps));
      await expect(Effect.runPromise(command)).rejects.toThrowErrorMatchingSnapshot(JSON.stringify({ _tag: TAG }));
      expect(ssmMock.calls().length).toEqual(1);
    });
  });
});
