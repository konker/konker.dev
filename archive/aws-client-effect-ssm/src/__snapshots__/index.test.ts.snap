// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aws-client-effect-ssm > DeleteParameterCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Name":"test-param-name"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-ssm > DeleteParametersCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Names":["test-param-name-1","test-param-name-2"]},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-ssm > GetParameterCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Name":"test-param-name"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-ssm > GetParametersByPathCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Path":"/foo/bar"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-ssm > GetParametersCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Names":["test-param-name-1","test-param-name-2"]},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;

exports[`aws-client-effect-ssm > PutParameterCommand > should fail as expected > {"_tag":"SsmError"} 1`] = `[(FiberFailure) Error: {"_tag":"SsmError","_Params":{"Name":"test-param-name-2","Type":"String","Value":"test-value-2"},"message":"","cause":{"$metadata":{"httpStatusCode":404}}}]`;
